core/variables.js

```diff
@@ -75,6 +76,22 @@ Blockly.Variables.allVariables = function(root) {
 };
 
 /**
+ * Find all variables that the user has created through the workspace or
+ * toolbox.  For use by generators.
+ * @param {!Blockly.Workspace} root The workspace to inspect.
+ * @return {!Array.<string>} Array of variable names.
+ */
+Blockly.Variables.allVariables = function(root) {
+  if (root.getDescendants) {
```

Comment:
Why not "root instanceof Blockly.Block"?  That way things don't get interesting if workspaces acquire a getDescendants function sometime in the future.